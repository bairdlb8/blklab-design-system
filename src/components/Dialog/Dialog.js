import React from 'react'
import PropTypes from 'prop-types'

import './Dialog.scss'

const Dialog = ({ className = '' }) => {
  return <div role="dialog">Dialog</div>
}

Dialog.propTypes = {}

export default Dialog
