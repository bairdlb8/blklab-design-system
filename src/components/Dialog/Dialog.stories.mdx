import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Dialog from './Dialog'

<Meta title="Components / Dialog" component={Dialog} />

# Dialog

## Add some text here

export const Template = (args) => {
  return <Dialog {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={{}}>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={Dialog} />
