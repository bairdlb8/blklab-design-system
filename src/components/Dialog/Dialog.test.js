import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import Dialog from './Dialog'

expect.extend(toHaveNoViolations)

describe('<Dialog />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Dialog />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Dialog />)

    expect(asFragment(<Dialog />)).toMatchSnapshot()
  })

})
