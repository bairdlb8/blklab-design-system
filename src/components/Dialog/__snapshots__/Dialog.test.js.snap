// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    role="dialog"
  >
    Dialog
  </div>
</DocumentFragment>
`;
